src/components/cc-addon-header/cc-addon-header.types.d.ts

```diff
@@ -0,0 +1,106 @@
+import { ZoneStateLoaded } from '../cc-zone/cc-zone.types.js';
+import { AddonPlan, AddonProvider } from '../common.types.js';
+
+export type CcAddonHeaderState =
+  | CcAddonHeaderStateLoading
+  | CcAddonHeaderStateLoaded
+  | CcAddonHeaderStateError
+  | CcAddonHeaderStateRestarting
+  | CcAddonHeaderStateRebuilding;
+
+interface BaseProperties {
+  providerName: string;
+  providerLogoUrl: string;
+  name: string;
+  id: string;
+  zone: ZoneStateLoaded;
+}
+
+interface OptionalProperties {
+  logsUrl?: string;
+  openLinks?: Array<OpenLink>;
+  actions?: {
+    restart: boolean;
+    rebuildAndRestart: boolean;
+  };
+  productStatus?: string;
+  deploymentStatus?: DeploymentStatus;
+  configLink?: string;
+}
+
+interface OpenLink {
+  url: string;
+  name: string;
+}
+
+export type DeploymentStatus = 'deploying' | 'active' | 'failed';
+
+export interface CcAddonHeaderStateLoading extends OptionalProperties {
+  type: 'loading';
+}
+
+export interface CcAddonHeaderStateLoaded extends BaseProperties, OptionalProperties {
+  type: 'loaded';
+}
+
+export interface CcAddonHeaderStateError {
+  type: 'error';
+}
+
+export interface CcAddonHeaderStateRestarting extends BaseProperties, OptionalProperties {
+  type: 'restarting';
+}
+
+export interface CcAddonHeaderStateRebuilding extends BaseProperties, OptionalProperties {
+  type: 'rebuilding';
+}
+
+export interface RawAddon {
+  id: string;
+  name: string;
+  realId: string;
+  region: string;
+  zoneId: string;
+  provider: AddonProvider;
+  plan: AddonPlan;
+  creationDate: number;
+  configKeys: string[];
+}
+
+export interface RawOperator {
+  resourceId: string;
+  addonId: string;
+  name: string;
+  ownerId: string;
+  plan: string;
+  version: string;
+  javaVersion: string;
+  accessUrl: string;
+  availableVersions: string[];
+  resources: {
+    entrypoint: string;
+    fsbucketId: string;
+    pgsqlId: string;
+  };
+  features: {
+    networkGroup?: string;
+  };
+  envVars: {
+    CC_KEYCLOAK_ADMIN: string;
+    CC_RUN_COMMAND: string;
+    CC_KEYCLOAK_DB_POOL_INITIAL_SIZE: string;
+    CC_METRICS_PROMETHEUS_PATH: string;
+    CC_JAVA_VERSION: string;
+    CC_PRE_RUN_HOOK: string;
+    CC_METRICS_PROMETHEUS_PORT: string;
+    CC_POST_BUILD_HOOK: string;
+    CC_KEYCLOAK_REALMS: string;
+    CC_KEYCLOAK_HOSTNAME: string;
+    CC_FS_BUCKET: string;
+    CC_KEYCLOAK_ADMIN_DEFAULT_PASSWORD: string;
+    CC_KEYCLOAK_VERSION: string;
+    CC_KEYCLOAK_DB_POOL_MAX_SIZE: string;
```

Comment:
fix: We do not need to be so precise in the env var types. Wecan simplify that to `Record<string, string>`